### PR TITLE
Include a project end date in the Reporting tab #2548

### PR DESCRIPTION
**Pull request by salomon-j**

### CODE REVIEW

File: grails-app/assets/javascripts/reporting.js [hunk 1]
```diff
@@ -580,32 +590,19 @@ var GrantManagerReportsViewModel = function(config) {
...
-            message =  "The project start date must be before the end date";
-        }
-
-        if (message) {
-            setTimeout(function() {
-                $(startDateSelector).validationEngine("showPrompt", message, "topRight", true);
-            }, 100);
-
-        } else {
+        $('#reportingTabDatesForm').validationEngine();
```
Comment (chrisala) [inline grails-app/assets/javascripts/reporting.js:593]: Please refactor the hardcoded selector here into a config item (e.g. config.datesFormSelector).

File: grails-app/assets/javascripts/reporting.js [hunk 2]
```diff
@@ -565,7 +565,17 @@ var CategorisedReportsViewModel = function(allReports, order, availableReports,
 var GrantManagerReportsViewModel = function(config) {
     var self = this;
     var projectService = new ProjectService(config.project, config);
-    self.plannedStartDate = ko.observable(config.reportOwner.startDate).extend({simpleDate: false});
+
+    var startDate;
+    var endDate;
+    if (config.project.status == 'Active') {
```
Comment (chrisala) [inline grails-app/assets/javascripts/reporting.js:571]: Refactor this into projectService.isActive(). (the project service has access to constants for active as well).

Comment (chrisala) [reply grails-app/assets/javascripts/reporting.js:571]: I am also a bit surprised about the capital "Active" refactoring this into the projectService could encapsulate a case insensitive comparision with the constant.  (The display value is "Active" but the value in the database should be "active").

**Review by chrisala — CHANGES_REQUESTED** [review]

File: grails-app/assets/javascripts/reporting.js [hunk 3]
```diff
@@ -580,32 +590,19 @@ var GrantManagerReportsViewModel = function(config) {
...
+            var jsData = {
+                plannedStartDate: self.plannedStartDate(),
+                plannedEndDate: self.plannedEndDate(),
+            };
             projectService.saveProjectData(jsData);
         }
     }
 
     self.isMeriPlanApproved = ko.pureComputed(function() {
-        if (!projectService.isApproved() && config.project.reports == '') {
```
[the remark this answers is not in the document]
Comment (salomon-j) [reply grails-app/assets/javascripts/reporting.js:608]: I think I'm still considering the legacy projects here, I will just remove the checking of the project report length as checking the meri plan status will suffice.

**Pull request merged**